plugin/src/com/microsoft/alm/plugin/idea/tfvc/core/TFVCNotifications.java

```diff
@@ -20,11 +24,28 @@
             true
     );
 
+    private static MergingUpdateQueue ourQueue = new MergingUpdateQueue(
+            "TFVCNotifications.myQueue",
+            2000,
+            false,
+            null,
+            null,
+            null,
+            true);
+
     public static void showInvalidDollarFilePathNotification(@NotNull Project project, @NotNull String serverFilePath) {
-        Notification notification = TFS_NOTIFICATIONS.createNotification(
-                TfPluginBundle.message(TfPluginBundle.KEY_TFVC_NOTIFICATION_FILE_NAME_STARTS_WITH_DOLLAR, serverFilePath),
-                NotificationType.WARNING);
-        notification.addAction(new AddFileToTfIgnoreAction(project, serverFilePath));
-        Notifications.Bus.notify(notification);
+        ourQueue.queue(new Update(Pair.create(project, serverFilePath)) {
```

Comment:
How many notifications would produce in this place?

Reply:
Usually they come in packs of three

Reply:
Ok, I'm afraid it may be not a good user experience. Maybe we could create a batch of files in one notification?

Reply:
No, you don't understand. Usually it creates three notifications about the same file :)

We don't know about any other dollar-containing files before we deal with the first one. Unfortunately, TF Everywhere won't tell us about any files except the fiest one.

Currently I don't believe this is a big issue, because most of the time there're only one or two files, and in other cases user may update the `.tfignore` manually.

Reply:
Ok, thank you for the explanation.